Update getting started with content management (#5196)
theforeman/foreman-documentation: guides/common/modules/proc_creating-a-product-and-a-repository.adoc

:_mod-docs-content-type: PROCEDURE

[id="creating-a-product-and-a-repository"]
= Creating a product and a repository

[role="_abstract"]
A product in {Project} groups repositories, and a repository in {Project} mirrors an upstream content source.

> Create a custom product and add a repository so that {Project} has content to synchronize.

Reviewer: ```suggestion
Create a {customproduct} and add a repository so that {Project} has content to synchronize.
```

See https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L70

ifdef::satellite[]
[NOTE]
====

> If you exclusively consume Red{nbsp}Hat content, proceed to xref:importing-a-subscription-manifest-and-enabling-red-hat-repositories[].

Reviewer: I'd slighty reword this to either "If your hosts exclusively consume ..." or "If you exclusively distribute RH content ...".

> If you exclusively consume Red{nbsp}Hat content, proceed to xref:importing-a-subscription-manifest-and-enabling-red-hat-repositories[].

Author: ```suggestion
If you exclusively distribute Red{nbsp}Hat content, proceed to xref:importing-a-subscription-manifest-and-enabling-red-hat-repositories[].
```

====
endif::[]

ifdef::katello[]
This example adds a repository with CentOS content.
endif::[]
ifdef::satellite[]
This example adds a repository with RPM content from EPEL (Extra Packages for Enterprise Linux).
endif::[]

.Prerequisites
* Your {ProjectWebUI} account has administrative privileges.

.Procedure
. In the {ProjectWebUI}, navigate to *Content* > *Products*.
. Click *Create Product*.
To configure the new product:
.. In the *Name* field, enter a name for the product.
ifdef::katello[]
For example: `CentOS Stream`.
endif::[]
ifdef::satellite[]
For example: `EPEL`.
endif::[]
.. Click *Save*.
. On the product page, click *New Repository*.
To configure the new repository:
.. In the *Name* field, enter a name for the repository.
ifdef::katello[]
For example: `BaseOS`.
endif::[]
ifdef::satellite[]
For example: `EPEL 9`.
endif::[]
.. From the *Type* list, select the required repository type.
For example: `yum`.

> .. In the *Upstream URL* field, enter the URL of the upstream repository.

Reviewer: Out of scope: As a follow-up, I will add a URL for orcharhino builds. -> I will create an issue after merging.

Reviewer: https://github.com/theforeman/foreman-documentation/pull/5243

ifdef::katello[]

> For example: `https://mirror.stream.centos.org/9-stream/BaseOS/x86_64/os/`.

Reviewer: ```suggestion
For example: `https://mirror.stream.centos.org/10-stream/BaseOS/x86_64/os/`.
```

I'd go with the latest available major version.

endif::[]
ifdef::satellite[]

> For example: `https://dl.fedoraproject.org/pub/epel/9/Everything/x86_64/`.

Reviewer: ```suggestion
For example: `https://dl.fedoraproject.org/pub/epel/10/Everything/x86_64/`.
```

endif::[]
.. From the *Download Policy* list, select a download policy.

> For example: Select *On Demand* to save disk space. With this policy, {Project} downloads only the metadata during synchronization and fetches the actual packages when a host requests them.

Reviewer: ```suggestion
For example: Select *On Demand* to save disk space.
With this policy, {Project} downloads only the metadata during synchronization and fetches the actual packages when a host requests them.
```

Author: Why didn't the OneSentencePerLine.yml rule catch this? I hope it's just due to Vale messages not being posted (a bug we're fixing elsewhere).

:_mod-docs-content-type: PROCEDURE

[id="creating-a-product-and-a-repository"]
= Creating a product and a repository

[role="_abstract"]
A product in {Project} groups repositories, and a repository in {Project} mirrors an upstream content source.
Create a custom product and add a repository so that {Project} has content to synchronize.

ifdef::satellite[]
[NOTE]
====

> If you only want to consume Red{nbsp}Hat content, proceed to xref:importing-a-subscription-manifest-and-enabling-red-hat-repositories[].

Reviewer 2: ```suggestion
If you exclusively consume Red{nbsp}Hat content, proceed to xref:importing-a-subscription-manifest-and-enabling-red-hat-repositories[].
```

====
endif::[]

ifdef::katello[]
This example adds a repository with CentOS content.
endif::[]
ifdef::satellite[]
This example adds a repository with RPM content from EPEL (Extra Packages for Enterprise Linux).
endif::[]

.Prerequisites
* Your {ProjectWebUI} account has administrative privileges.

.Procedure
. In the {ProjectWebUI}, navigate to *Content* > *Products*.
. Click *Create Product*.
To configure the new product:
.. In the *Name* field, enter a name for the product.
ifdef::katello[]
For example: `CentOS Stream`.
endif::[]
ifdef::satellite[]
For example: `EPEL`.
endif::[]
.. Click *Save*.
. On the product page, click *New Repository*.
To configure the new repository:
.. In the *Name* field, enter a name for the repository.
ifdef::katello[]
For example: `BaseOS`.
endif::[]
ifdef::satellite[]
For example: `EPEL 9`.
endif::[]
.. From the *Type* list, select the required repository type.
ifdef::katello[]
For example: `yum`.
endif::[]
ifdef::satellite[]
For example: `yum`.

> endif::[]

Reviewer 2: ```suggestion
For example: `yum`.
```

.. In the *Upstream URL* field, enter the URL of the upstream repository.
ifdef::katello[]
For example: `https://mirror.stream.centos.org/9-stream/BaseOS/x86_64/os/`.
endif::[]
ifdef::satellite[]
For example: `https://dl.fedoraproject.org/pub/epel/9/Everything/x86_64/`.
endif::[]
.. From the *Download Policy* list, select *On Demand* to save disk space.

> With this policy, {Project} downloads only the metadata during synchronization and fetches the actual packages when a host requests them.

Reviewer 2: I think something like this makes a little more sense but feel free to reword.
```suggestion
.. From the *Download Policy* list, select a download policy.
For example: Select *On Demand* to save disk space. With this policy, {Project} downloads only the metadata during synchronization and fetches the actual packages when a host requests them.
```
